spark/src/main/scala/org/apache/spark/sql/delta/commands/merge/MergeOutputGeneration.scala

```diff
@@ -215,69 +212,74 @@ trait MergeOutputGeneration { self: MergeIntoCommandBase =>
    * @param cdcEnabled Whether the generated expressions should include CDC information.
    * @param shouldCountDeletedRows Whether metrics for number of deleted rows should be incremented
    *                               here.
-   * @return For each merge clause, a list of [[ProcessedClause] each with a precomputed
+   * @return For each merge clause, a list of [[ProcessedClause]] each with a precomputed
    *         condition and N+2 action expressions (N output columns + [[ROW_DROPPED_COL]] +
    *         [[CDC_TYPE_COLUMN_NAME]]) to apply on a row when that clause matches.
    */
   protected def generateAllActionExprs(
       targetOutputCols: Seq[Expression],
       clausesWithPrecompConditions: Seq[DeltaMergeIntoClause],
       cdcEnabled: Boolean,
-      shouldCountDeletedRows: Boolean)
-    : Seq[ProcessedClause] = clausesWithPrecompConditions.map (clause => {
-    val incrUpdatedCountExpr =
-      incrementMetricAndReturnBool(name = "numTargetRowsUpdated", valueToReturn = false)
-    val incrDeletedCountExpr =
-      incrementMetricAndReturnBool(name = "numTargetRowsDeleted", valueToReturn = true)
-
-    val actions = clause match {
-      // Seq of up to N+3 expressions to generate output rows based on the UPDATE, DELETE and/or
-      // INSERT action(s)
-      case u: DeltaMergeIntoMatchedUpdateClause =>
-        val incrCountExpr =
-          IncrementMetric(incrUpdatedCountExpr, metrics("numTargetRowsMatchedUpdated"))
```

Comment:
Looks like `IncrementMetric` isn't used anymore and the import can be removed

Reply:
Good catch! Fixed.